server/src/test/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryTests.java

```diff
@@ -119,50 +124,16 @@ public void testUploadedSegmentMetadataFromString() {
         assertEquals("_0.cfe::_0.cfe__uuidxyz::4567::372000", metadata.toString());
     }
 
-    public void testGetMetadataFilename() {
-        // Generation 23 is replaced by n due to radix 32
-        assertEquals(
-            RemoteSegmentStoreDirectory.MetadataFilenameUtils.METADATA_PREFIX + "__12__n__uuid1",
-            RemoteSegmentStoreDirectory.MetadataFilenameUtils.getMetadataFilename(12, 23, "uuid1")
-        );
-    }
-
     public void testGetPrimaryTermGenerationUuid() {
-        String[] filenameTokens = "abc__12__n__uuid_xyz".split(RemoteSegmentStoreDirectory.MetadataFilenameUtils.SEPARATOR);
+        String[] filenameTokens = "abc__9223372036854775795__9223372036854775784__uuid_xyz".split(
+            RemoteSegmentStoreDirectory.MetadataFilenameUtils.SEPARATOR
+        );
         assertEquals(12, RemoteSegmentStoreDirectory.MetadataFilenameUtils.getPrimaryTerm(filenameTokens));
         assertEquals(23, RemoteSegmentStoreDirectory.MetadataFilenameUtils.getGeneration(filenameTokens));
-        assertEquals("uuid_xyz", RemoteSegmentStoreDirectory.MetadataFilenameUtils.getUuid(filenameTokens));
-    }
-
-    public void testMetadataFilenameComparator() {
```

Comment:
we should add tests to assert md files are reverse lexicographically sorted by primary term and generation .